pkg/kubernetes/statefulset.go

```diff
@@ -0,0 +1,109 @@
+package kubernetes
+
+import (
+	"encoding/json"
+	//"fmt"
+	"reflect"
+	"strings"
+	"github.com/billiford/go-clouddriver/pkg/kubernetes/manifest"
+	v1 "k8s.io/api/apps/v1"
+)
+
+type StatefulSet interface {
+	Object() *v1.StatefulSet
+	SetReplicas(*int32)
+	Status() manifest.Status
+}
+
+type statefulSet struct {
+	ss *v1.StatefulSet
+}
+
+func NewStatefulSet(m map[string]interface{}) StatefulSet {
+	s := &v1.StatefulSet{}
+	b, _ := json.Marshal(m)
+	_ = json.Unmarshal(b, &s)
+
+	return &statefulSet{ss: s}
+}
+
+func (ss *statefulSet) Object() *v1.StatefulSet {
+	return ss.ss
+}
+
+func (ss *statefulSet) SetReplicas(replicas *int32) {
+	ss.ss.Spec.Replicas = replicas
+}
+
+func (ss *statefulSet) Status() manifest.Status {
+	s := manifest.DefaultStatus
+	x := ss.ss
+
+	if strings.EqualFold(string(x.Spec.UpdateStrategy.Type), "ondelete") {
+		return s
+	}
+
+	if reflect.DeepEqual(x.Status, v1.StatefulSetStatus{}) {
+		s = manifest.NoneReported
+
+		return s
+	}	
+
+	if x.ObjectMeta.Generation != x.Status.ObservedGeneration {
+		s.Stable.State = false
+		s.Stable.Message = "Waiting for status generation to match updated object generation"
+		return s
+	}
+
+	desired := int32(0)
+	if x.Spec.Replicas != nil {
+		desired = *x.Spec.Replicas
+	}
+
+	existing := x.Status.Replicas
+	if desired > existing {
+		s.Stable.State = false
+		s.Stable.Message = "Waiting for at least the desired replica count to be met"
+
+		return s
+	}
+
+	ready := x.Status.ReadyReplicas
+	if desired > ready {
+		s.Stable.State = false
+		s.Stable.Message = "Waiting for all updated replicas to be ready"
+
+		return s
+	}
+
+	updType := string(x.Spec.UpdateStrategy.Type)	
+	updated := x.Status.UpdatedReplicas
+	partition := *x.Spec.UpdateStrategy.RollingUpdate.Partition
+	
+	if strings.EqualFold(updType, "rollingupdate") && updated != 0 && partition != 0 {
```

Comment:
In the source there is no check that `partition != 0` in this if statement, instead it is a check that [rollingUpdate != null](https://github.com/spinnaker/clouddriver/blob/master/clouddriver-kubernetes/src/main/java/com/netflix/spinnaker/clouddriver/kubernetes/op/handler/KubernetesStatefulSetHandler.java#L149).

If `x.Spec.UpdateStrategy.RollingUpdate` is not a pointer then we don't have to check this.